client/src/helpers/getOptionalAttributes.js

```diff
@@ -0,0 +1,18 @@
+// converts a schema to a list of optional attributes
+export default (schema, object) => {
+  const optionalAttributes = []
+
+  const addRequired = ([key, desc]) => {
```

Comment:
Is there a reason this takes a list and deconstructs it instead of just taking the args directly?

Reply:
Yeah this is more of a design choice since its defined inside the helper and used recursively I can do this which passes the pair in.
I prefer this:
```javascript
    if (desc.fields) Object.entries(desc.fields).forEach(addRequired)
```
Over this:
```javascript
    if (desc.fields) Object.entries(desc.fields).forEach(([k, v]) => addRequired(k, v))
```
```